File: Java/1266_Minimum_Time_Visiting_All_Points.java
class Solution {
    public int minTimeToVisitAllPoints(int[][] points) {
        int result = 0;
        int x = points[0][0];
        int y = points[0][1];
        
        for(int i = 1; i < points.length; i++){
            int tx = points[i][0];
            int ty = points[i][1];
            
            result += Math.max(Math.abs(x-tx), Math.abs(y-ty));
            x = tx;
            y = ty;
        }
        
        return result;
    }
}

File: CS/1266_Minimum_Time_Visiting_All_Points.cs
public class Solution {
    public int MinTimeToVisitAllPoints(int[][] points) {
        int result = 0;
        
        for(int i = 1; i < points.Length; i++){
            result += Math.Max(Math.Abs(points[i][0] - points[i-1][0]), Math.Abs(points[i][1] - points[i-1][1]));
        }
        
        return result;
    }
    /*
    public int MinTimeToVisitAllPoints(int[][] points) =>
        Enumerable.Range(1, points.Length - 1)
            .Select(i => Math.Max(Math.Abs(points[i][0] - points[i - 1][0]), Math.Abs(points[i][1] - points[i - 1][1])))
            .Sum();
}
    */
}
